scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -309,6 +310,11 @@ def handle_modules(self):
         modules_meta = parse_modules(ZEPHYR_BASE)
         self.modules = [module.meta.get('name') for module in modules_meta]
 
+        # Check each module for a snippets directory, then add to a list if it is found
+        for module in modules_meta:
+            snippet_path = module.project + '/snippets'
+            if os.path.exists(snippet_path):
+                self.snippet_roots.append(Path(module.project))
```

Comment:
We should not have a deviating implementation in twister where snippets are picked up differently than in the build system.

The build system uses the information provided in `zephyr/module.yml` but this code checks for the existence of a folder regardless of the setting in `zephyr/module.yml`.